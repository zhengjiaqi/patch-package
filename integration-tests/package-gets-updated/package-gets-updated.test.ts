import { runIntegrationTest } from "../runIntegrationTest"
runIntegrationTest({
  projectName: "package-gets-updated",
  shouldProduceSnapshots: true,
})
