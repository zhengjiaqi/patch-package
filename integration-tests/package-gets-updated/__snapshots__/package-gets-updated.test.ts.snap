// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test package-gets-updated: fail when the patch was not applied 1`] = `
"SNAPSHOT: fail when the patch was not applied
warning left-pad@1.1.3: use String.prototype.padStart()

**ERROR** Failed to apply patch for package left-pad at path
  
    node_modules/left-pad

  This error was caused because left-pad has changed since you
  made the patch file for it. This introduced conflicts with your patch,
  just like a merge conflict in Git when separate incompatible changes are
  made to the same piece of code.

  Maybe this means your patch file is no longer necessary, in which case
  hooray! Just delete it!

  Otherwise, you need to generate a new patch file.

  To generate a new one, just repeat the steps you made to generate the first
  one.

  i.e. manually make the appropriate file changes, then run 

    patch-package left-pad

  Info:
    Patch file: patches/left-pad+1.1.1.patch
    Patch was made for version: 1.1.1
    Installed version: 1.1.3

---
patch-package finished with 1 error(s).
error Command failed with exit code 1.
END SNAPSHOT"
`;

exports[`Test package-gets-updated: left-pad should contain patch-package 1`] = `
"SNAPSHOT: left-pad should contain patch-package
    // devide \`len\` by 2, ditch the patch-package
END SNAPSHOT"
`;

exports[`Test package-gets-updated: left-pad should still contain patch-package 1`] = `
"SNAPSHOT: left-pad should still contain patch-package
    // devide \`len\` by 2, ditch the patch-package
END SNAPSHOT"
`;

exports[`Test package-gets-updated: warning when the patch was applied but version changed 1`] = `
"SNAPSHOT: warning when the patch was applied but version changed

Warning: patch-package detected a patch file version mismatch

  Don't worry! This is probably fine. The patch was still applied
  successfully. Here's the deets:

  Patch file created for

    left-pad@1.1.1

  applied to

    left-pad@1.1.2
  
  At path
  
    node_modules/left-pad

  This warning is just to give you a heads-up. There is a small chance of
  breakage even though the patch was applied successfully. Make sure the package
  still behaves like you expect (you wrote tests, right?) and then run

    patch-package left-pad

  to update the version in the patch file name and make this warning go away.

---
patch-package finished with 1 warning(s).
END SNAPSHOT"
`;
